### PR TITLE
Eql backend

### PR DESCRIPTION
**Pull request by thomaspatzke**

### CODE REVIEW

File: sigma/backends/elasticsearch/__init__.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
-from .elasticsearch import LuceneBackend
+from .elasticsearch_lucene import LuceneBackend
+from .elasticsearch_eql import EqlBackend
 
 backends = {
```
Comment (thomaspatzke) [inline sigma/backends/elasticsearch/__init__.py:4]: This is not really required anymore with pySigma 0.10 autodiscovery feature.

File: sigma/backends/elasticsearch/elasticsearch_eql.py [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+    ConditionFieldEqualsValueExpression,
+)
+from sigma.types import SigmaCompareExpression, SigmaNull, SpecialChars, SigmaNumber
+import ipaddress
+import sigma
+
+
+class EqlBackend(TextQueryBackend):
+    """
+    Elasticsearch event querly language backend. Generates query strings described here in the
```
Comment (thomaspatzke) [inline sigma/backends/elasticsearch/elasticsearch_eql.py:23]: s/querly/query/

File: sigma/backends/elasticsearch/elasticsearch_eql.py [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+
+    # A descriptive name of the backend
+    name: ClassVar[str] = "Elasticsearch EQL"
+    # Output formats provided by the backend as name -> description mapping.
+    # The name should match to finalize_output_<name>.
+    formats: ClassVar[Dict[str, str]] = {
+        "default": "Plain Elasticsearch EQL queries.",
+        "eqlapi": "Plain EQL queries ready for '/_eql/search' API endpoint.",
+        "siem_rule": "Elasticsearch EQL queries as SIEM Rule.",
+        "siem_rule_ndjson": "Elasticsearch EQL Query as SIEM Rules in NDJSON Format.",
```
Comment (thomaspatzke) [inline sigma/backends/elasticsearch/elasticsearch_eql.py:37]: Perhaps something that could be replaced with post-processing feature from pySigma >= 0.10.

File: sigma/backends/elasticsearch/elasticsearch_eql.py [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+                self.wildcard_match_expression is not None
+                and cond.value.contains_special()
+            ):
+                expr = self.wildcard_match_expression
+                value = cond.value
+            elif (  # Use '==' as operator for empty string or ip addresses
+                cond.value.convert() == "" or self.is_ip(cond.value)
+            ):
+                expr = "{field}" + "==" + "{value}"
+                value = cond.value
```
Comment (thomaspatzke) [inline sigma/backends/elasticsearch/elasticsearch_eql.py:245]: Except this if-statement the code is copy&paste from the backend base class. The override should better only contain this if block that also incorporates the return statement at the botton and an else block that returns the result of the base class implementation called with `super()`.

File: sigma/backends/elasticsearch/elasticsearch_eql.py [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+    group_expression: ClassVar[str] = "({expr})"
+    parenthesize: bool = True
+
+    # Generated query tokens
+    token_separator: str = " "  # separator inserted between all boolean operators
+    or_token: ClassVar[str] = "or"
+    and_token: ClassVar[str] = "and"
+    not_token: ClassVar[str] = "not"
+    # Token inserted between field and value (without separator)
+    eq_token: ClassVar[str] = ":"
```
Comment (thomaspatzke) [inline sigma/backends/elasticsearch/elasticsearch_eql.py:59]: pySigma 0.10 added support for case-sensitive matching that could be integrated here with the `==` operator. See [base backend code](https://github.com/SigmaHQ/pySigma/blob/485773b0557d334a006a45a5b046fbe22bd9a596/sigma/conversion/base.py#L628) for more details on this.

File: sigma/backends/elasticsearch/__init__.py (hunk 1 above)
Comment (andurin) [reply sigma/backends/elasticsearch/__init__.py:4]: But at least for the local tests which don't participate in the autodiscovery. 
Am I wrong?

File: sigma/backends/elasticsearch/elasticsearch_eql.py (hunk 2 above)
Comment (andurin) [reply sigma/backends/elasticsearch/elasticsearch_eql.py]: Fixed.

File: sigma/backends/elasticsearch/__init__.py (hunk 1 above)
Comment (thomaspatzke) [reply sigma/backends/elasticsearch/__init__.py:4]: For these you should instantiate the backend directly.

**Comment by andurin** [comment]
@thomaspatzke check_code_happiness()
:-)

File: sigma/backends/elasticsearch/elasticsearch_eql.py (hunk 3 above)
Comment (andurin) [reply sigma/backends/elasticsearch/elasticsearch_eql.py:37]: I'll do that as a seperate issue - https://github.com/SigmaHQ/pySigma-backend-elasticsearch/issues/33

**Pull request merged**